File: TrackBudget/TrackBudget.Domain/Entities/Transaction.cs
using TrackBudget.Domain.Common;
using TrackBudget.Domain.Enums;

namespace TrackBudget.Domain.Entities;

public class Transaction : BaseEntity
{
    public string Title { get; set; } = string.Empty;

    public decimal Amount { get; set; }

    public TransactionType Type { get; set; }

    public DateTime Date { get; set; }
    public string? Notes { get; set; }

    public Guid UserId { get; set; }

    public User User { get; set; } = null!;

    public Guid AccountId { get; set; }

    public Account Account { get; set; } = null!;

    public Guid CategoryId { get; set; }

    public Category Category { get; set; } = null!;

}

File: TrackBudget/TrackBudget.Client/src/types/transaction.ts
export type TransactionType = "Income" | "Expense";

export interface Transaction {
  id: string;
  title: string;
  amount: number;
  accountId: string;
  accountName: string;
  categoryId: string;
  categoryName?: string;
  date: string;
  notes?: string;
  type: TransactionType;
  createdAt: string;
}

export interface CreateTransactionRequest {
  title: string;
  amount: number;
  accountId: string;
  categoryId: string;
  date: string;
  notes?: string;
  type: TransactionType;
}

export interface UpdateTransactionRequest {
  title?: string;
  amount?: number;
  accountId?: string;
  categoryId?: string;
  date?: string;
  notes?: string;
  type?: TransactionType;
}
